### PR TITLE
WFLY-4515 Fix clustering transformers and reenable in mixed-domain tests (Infinispan part)

### PR DESCRIPTION
**Pull request by rhusar**
Jira
https://issues.jboss.org/browse/WFLY-4515

Fix multiple Infinispan transformations
- fix SimpleAttributeConverter failing with ResourceNotFound
- fix ImplicitlyAddedResourceDynamicDiscardPolicy
- fix missing child redirections (e.g. binary-keyed store)
- fix table tranformations for all store resource definitions
- fix value conversion for EvictionResourceDefinition and TransactionResourceDefinition
- fix outbound-socket-binding conversion in RemoteStoreResourceDefinition
- fix validationFailedRequiredParameterNotPresent not thrown

Test fixes
- reenable mixed-domain tests for infinispan subystem
- add tests for EAP 6.4
- add missing capabilities for tests
- make test xml files pass validation
- fix redundant modifiers
- add missing javadoc

Fix properties transformation in Infinispan subsystem
- fix properties transformation for all stores
- extract properties transformations common for jgroups and infinispan subsystem
- simplify transformation registration for all write operations

### CODE REVIEW

File: testsuite/mixed-domain/src/test/java/org/jboss/as/test/integration/domain/mixed/eap640/DomainAdjuster640.java [hunk 1]
```diff
@@ -188,6 +186,41 @@
     }
 
 
+    private List<ModelNode> adjustInfinispan(final PathAddress subsystem) throws Exception {
+        final List<ModelNode> list = new ArrayList<>();
+
+        list.add(getWriteAttributeOperation(subsystem.append("cache-container", "server").append("transport", "jgroups"), "stack", new ModelNode("udp")));
+
+        //Statistics need to be enabled for all cache containers and caches
+        list.add(setStatisticsEnabledTrue(
```
Comment (kabir) [inline testsuite/mixed-domain/src/test/java/org/jboss/as/test/integration/domain/mixed/eap640/DomainAdjuster640.java]: Statistics-enabled were introduced in 6.4.0, so there is no need to set them to true for 6.4.0, only for 6.3.0

**Comment by kabir** [comment]
Although not part of this PR, I see in CacheContainerResourceDefinition
{code}
        if (InfinispanModel.VERSION_1_5_0.requiresTransformation(version)) {
            builder.getAttributeBuilder()
                    // discard statistics if set to true, reject otherwise
                    .setDiscard(new DiscardAttributeChecker.DiscardAttributeValueChecker(false, false, new ModelNode(true)), Attribute.STATISTICS_ENABLED.getDefinition())
                    .addRejectCheck(RejectAttributeChecker.UNDEFINED, Attribute.STATISTICS_ENABLED.getDefinition())
                    .addRejectCheck(RejectAttributeChecker.SIMPLE_EXPRESSIONS, Attribute.STATISTICS_ENABLED.getDefinition()) <=====
{code}
this attribute was introduced in 6.4.0, and expressions were allowed then (as they are now)

File: clustering/infinispan/extension/src/test/java/org/jboss/as/clustering/infinispan/subsystem/TransformersTestCase.java [hunk 2]
```diff
@@ -263,8 +290,10 @@ private static FailedOperationTransformationConfig createFailedOperationConfig(M
         if (InfinispanModel.VERSION_2_0_0.requiresTransformation(version)) {
             for (PathElement path : Arrays.asList(DistributedCacheResourceDefinition.WILDCARD_PATH, ReplicatedCacheResourceDefinition.WILDCARD_PATH)) {
                 PathAddress cacheAddress = containerAddress.append(path);
-                config.addFailedAttribute(cacheAddress.append(BackupResourceDefinition.WILDCARD_PATH), FailedOperationTransformationConfig.REJECTED_RESOURCE);
-                config.addFailedAttribute(cacheAddress.append(BackupForResourceDefinition.PATH), FailedOperationTransformationConfig.REJECTED_RESOURCE);
+                config.addFailedAttribute(
+                        cacheAddress.append(BackupsResourceDefinition.PATH).append(BackupResourceDefinition.WILDCARD_PATH), FailedOperationTransformationConfig.REJECTED_RESOURCE);
+                config.addFailedAttribute(
+                        cacheAddress.append(BackupForResourceDefinition.PATH), FailedOperationTransformationConfig.REJECTED_RESOURCE);
```
Comment (kabir) [inline clustering/infinispan/extension/src/test/java/org/jboss/as/clustering/infinispan/subsystem/TransformersTestCase.java]: The reject xml should not use statistics-enabled=true, and this should be added as a failed attribute in the relevant cases.

**Comment by kabir** [comment]
Have summarised changes that need to be made in private with Rado

File: clustering/infinispan/extension/src/test/java/org/jboss/as/clustering/infinispan/subsystem/TransformersTestCase.java (hunk 2 above)
Comment (rhusar) [inline clustering/infinispan/extension/src/test/java/org/jboss/as/clustering/infinispan/subsystem/TransformersTestCase.java]: Done.

File: testsuite/mixed-domain/src/test/java/org/jboss/as/test/integration/domain/mixed/eap640/DomainAdjuster640.java (hunk 1 above)
Comment (rhusar) [inline testsuite/mixed-domain/src/test/java/org/jboss/as/test/integration/domain/mixed/eap640/DomainAdjuster640.java]: Done.

**Pull request merged**